packages/wonder-blocks-data/src/util/__tests__/memory-cache.test.js

```diff
@@ -267,26 +284,80 @@ describe("MemoryCache", () => {
                 hydrate: true,
             };
 
+            // Act
+            cache.removeAll(fakeHandler, (k, d) => d.data === "2");
+            const result = cache.cloneData();
+
+            // Assert
+            expect(result).toStrictEqual({
+                MY_HANDLER: {
+                    MY_KEY2: {data: "1"},
+                },
+                OTHER_HANDLER: {
+                    MY_KEY: {data: "1"},
+                },
+            });
+        });
+
+        it("should return the number of items that matched the predicate and were removed", () => {
+            const cache = new MemoryCache({
+                MY_HANDLER: {
+                    MY_KEY: {data: "2"},
+                    MY_KEY2: {data: "1"},
+                    MY_KEY3: {data: "2"},
+                },
+                OTHER_HANDLER: {
+                    MY_KEY: {data: "1"},
+                },
+            });
+            const fakeHandler: IRequestHandler<string, string> = {
+                getKey: () => "MY_KEY",
+                type: "MY_HANDLER",
+                fulfillRequest: jest.fn(),
+                hydrate: true,
+            };
+
             // Act
             const result = cache.removeAll(
                 fakeHandler,
                 (k, d) => d.data === "2",
             );
-            const after = cache.cloneData();
 
             // Assert
             expect(result).toBe(2);
```

Comment:
style nit: using `"2"` for the data and the result also being `2` might make people think they're related even though they're not.

Reply:
Oh yeah, nice. I'll fix that.